prototype_source/lite_interpreter.rst

```diff
@@ -0,0 +1,221 @@
+(Prototype) Introduce lite interpreter workflow in Android and iOS
+==================================================================
+
+**Author**: `Chen Lai <https://github.com/cccclai>`_, `Martin Yuan <https://github.com/iseeyuan>`_
+
+Introduction
+------------
+
+This tutorial introduces the steps to use lite interpreter on iOS and Android. We'll be using the ImageSegmentation demo app as an example. Since lite interpreter is currently in the prototype stage, a custom pytorch binary from source is required.
+
+
+Android
+-------------------
+Get ImageSegmentation demo app in Android: https://github.com/pytorch/android-demo-app/tree/master/ImageSegmentation
+
+1. **Prepare model**: Prepare the lite interpreter version of model by run the script below to generate the scripted model `deeplabv3_scripted.pt` and `deeplabv3_scripted.ptl`
+
+.. code:: python
+
+    import torch
+
+    model = torch.hub.load('pytorch/vision:v0.7.0', 'deeplabv3_resnet50', pretrained=True)
+    model.eval()
+
+    scripted_module = torch.jit.script(model)
+    # Export full jit version model (not compatible lite interpreter), leave it here for comparison
+    scripted_module.save("deeplabv3_scripted.pt")
+    # Export lite interpreter version model (compatible with lite interpreter)
+    scripted_module._save_for_lite_interpreter("deeplabv3_scripted.ptl")
+
+2. **Build libtorch lite for android**: Build libtorch for android for all 4 android abis (``armeabi-v7a``, ``arm64-v8a``, ``x86``, ``x86_64``) ``BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh``. For example, if it will be tested on Pixel 4 emulator with ``x86``, use cmd ``BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`` to specify abi to save built time. After the build finish, it will show the library path:
+
+
+.. code-block:: bash
+
+   BUILD SUCCESSFUL in 55s
+   134 actionable tasks: 22 executed, 112 up-to-date
+   + find /Users/chenlai/pytorch/android -type f -name '*aar'
+   + xargs ls -lah
+   -rw-r--r--  1 chenlai  staff    13M Feb 11 11:48 /Users/chenlai/pytorch/android/pytorch_android/build/outputs/aar/pytorch_android-release.aar
+   -rw-r--r--  1 chenlai  staff    36K Feb  9 16:45 /Users/chenlai/pytorch/android/pytorch_android_torchvision/build/outputs/aar/pytorch_android_torchvision-release.aar
+
+3. **Use the PyTorch Android libraries built from source in the ImageSegmentation app**: Create a folder `libs` in the path, the path from repository root will be `ImageSegmentation/app/libs`. Copy `pytorch_android-release` to the path ``ImageSegmentation/app/libs/pytorch_android-release.aar``. Copy `pytorch_android_torchvision` (downloaded from `Pytorch Android Torchvision Nightly <https://oss.sonatype.org/#nexus-search;quick~torchvision_android/>`_) to the path ``ImageSegmentation/app/libs/pytorch_android_torchvision.aar``. Update the `dependencies` part of ``ImageSegmentation/app/build.gradle`` to
```

Comment:
Is it using "libraries from nightly build", instead of "from source"? Could we add "nightly build" specifically in the title, since it would be the suggested way for users to try lite interpreter?

Reply:
It's for Android nightly build. How about iOS?

Reply:
> Is it using "libraries from nightly build", instead of "from source"? Could we add "nightly build" specifically in the title, since it would be the suggested way for users to try lite interpreter?

It is only for torch vision, since I ran into some issues if I use the official one. 

> It's for Android nightly build. How about iOS?

It is a workaround to make lite interpreter built from source to work in Android. The workaround is to use torch vision from nightly, instead of the official one. iOS doesn't need this.

Reply:
Very good feedback. I am trying to think what is the best way to describe it..